modules/op-mitigating-extra-pipeline-resource-consumption.adoc

```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// */openshift_pipelines/uninstalling-pipelines.adoc
+
+[id='op-mitigating-extra-pipeline-resource-consumption_{context}']
+= Mitigating extra resource consumption in pipelines
+
+When you have resource limits set on the container of your pod, {product-title} sums up the resource limits requested as all containers run at the same time.
+
+To consume the bare minimum amount of resources needed to execute one step at a time from the invoked task, {pipelines-title} requests the maximum CPU, memory, and ephemeral storage as specified in the step that requires the most amount of resources. This ensures that the resource requirements of the step that needs the most amount of resources as well as the steps that need a lesser amount of resources are met. Requests other than the maximum values are set to zero.
+
+However, this behavior can lead to higher resource usage than required. If you use resource quotas, this could also lead to unschedulable pods.
+
+For example, in a task that has two steps that uses scripts and that does not define any resource limits and requests.
+The resulting pod will have two init containers (one for entrypoint copy, the other for writing scripts) and two containers, one for each step.
+
+For this example, use the following limit range in the project:
+
+----
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: mem-min-max-demo-lr
+spec:
+  limits:
+  - max:
+      memory: 1Gi
+    min:
+      memory: 500Mi
+    type: Container
+----
+
+{product-title} uses the limit range set up for the project to compute required resource requests and limits. In this example, each init container would use a request memory of 1Gi (the max limit of the limit range) and each container would use a request memory of 500Mi. Thus the total memory request for the pod would be 2Gi.
```

Comment:
{product-title} uses the limit range set for the project to compute required resource requests and limits. In this example, each `init` container requests memory of 1Gi (the maximum permissible value in the limit range), and each container would use a request memory of 500Mi. Thus, the total memory request for the pod would be 2Gi.

Reply:
Not sure what is the suggestion here.

Reply:
A couple of commas - I thought rather thyaying saying "put a comma here, here, and here", it would be more efficient if I just give my version.

Reply:
Ah, ok.